tc/core/polyhedral/memory_promotion.cc

```diff
@@ -96,6 +96,11 @@ std::pair<isl::val, isl::aff> outputRangeSingle(isl::map access) {
 // TODO: replace this with a call to isl_map_range_box_hull when available.
 ScopedFootprint outputRanges(isl::map access) {
   int nSubscripts = access.dim(isl::dim_type::out);
+
+  if (access.get_space().domain().is_wrapping()) {
+    access = access.domain_factor_domain();
+  }
```

Comment:
This is too much magic.
It should be the callers responsibility to ensure consistency.
See f183f2ed242b76960975944eba7576f09b326193 in #477.